### PR TITLE
realtime chat

### PR DESCRIPTION
**Pull request by wowKimsohyun17**
## Related Tickets:
- [#32626](https://edu-redmine.sun-asterisk.vn/issues/32626)

## WHAT this PR does?
Write controller, event to process realtime chat

## Notes Impacted Areas:
- N/A.

## Attachments:
peek-test-chat

### CODE REVIEW

**Comment by wowKimsohyun17** [comment]
ready

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -6,39 +6,59 @@
...
-        $course->load(['image', 'lessons', 'comments.user.image']);
+        $course = $this->courseRepo->loadRelations($course, ['image', 'lessons', 'comments.user.image']);
         
         return view('user.component.lessons', compact('course'));
     }
 
     public function showMentors()
     {   
-        $mentors = User::with('mentorComments.user.image', 'image')->where('role_id', config('role.mentor_id'))->get();
+        $mentors = $this->userRepo->getAllWithWhere(['mentorComments.user.image', 'image'], ['role_id' => config('role.mentor_id')]);
```
Comment (huyenttm-0901) [inline app/Http/Controllers/HomeController.php]:
```suggestion
        $mentors = $this->userRepo->getData(['mentorComments.user.image', 'image'], ['role_id' => config('role.mentor_id')]);
```
sửa lại tên function nhé

File: app/Http/Controllers/User/StudentController.php [hunk 2]
```diff
@@ -171,11 +171,15 @@ public function storeLessonComment(StoreCommentRequest $request, Lesson $lesson)
...
     {
-        Advisor::create([
+        $data = [
             'lesson_id' => $request->lesson_id,
             'student_id' => Auth::id(),
             'status' => config('status.request.pending_number'),
-        ]);
+        ];
+        $advisor = Advisor::where($data)->get();
+        if (!$advisor->count() > 0) {
```
Comment (huyenttm-0901) [inline app/Http/Controllers/User/StudentController.php]:
```suggestion
        if (!$advisor) {
```

**Comment by wowKimsohyun17** [comment]
ready

**Pull request merged**